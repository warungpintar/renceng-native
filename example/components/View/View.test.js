import React from 'react';
import {render} from '@testing-library/react-native';
import {View} from '@warungpintar/renceng-native';

describe('View', () => {
  test('basic', () => {
    const props = {
      testID: 'my-view',
      width: 200,
      height: 100,
      bg: 'white',
      borderWidth: 1,
      borderColor: 'black',
      borderStyle: 'solid',
    };
    const {queryByTestId, toJSON} = render(<View {...props}> </View>);
    const style = queryByTestId('my-view').props.style[0];
    expect(style.backgroundColor).toMatch('white');
    expect(style.borderColor).toMatch('black');
    expect(style.borderStyle).toMatch('solid');
    expect(style.borderWidth).toEqual(1);
    expect(style.height).toEqual(100);
    expect(style.width).toEqual(200);
    expect(toJSON()).toMatchSnapshot();
  });

  test('rounded', () => {
    const props = {
      testID: 'my-view',
      width: 200,
      height: 100,
      bg: 'white',
      borderWidth: 1,
      borderColor: 'black',
      borderStyle: 'solid',
      borderRadius: 8,
    };
    const {queryByTestId, toJSON} = render(<View {...props}> </View>);
    const style = queryByTestId('my-view').props.style[0];
    expect(style.backgroundColor).toMatch('white');
    expect(style.borderColor).toMatch('black');
    expect(style.borderStyle).toMatch('solid');
    expect(style.borderWidth).toEqual(1);
    expect(style.borderRadius).toEqual(8);
    expect(style.height).toEqual(100);
    expect(style.width).toEqual(200);
    expect(toJSON()).toMatchSnapshot();
  });

  test('shadow', () => {
    const props = {
      testID: 'my-view',
      width: 200,
      height: 100,
      bg: 'white',
      shadowColor: 'black',
      shadowOffset: {
        width: 4,
        height: 8,
      },
      shadowOpacity: 0.5,
      shadowRadius: 8,
      elevation: 8,
    };
    const {queryByTestId, toJSON} = render(<View {...props}> </View>);
    const style = queryByTestId('my-view').props.style[0];
    const elementProps = queryByTestId('my-view').props;
    expect(style.backgroundColor).toMatch('white');
    expect(style.height).toEqual(100);
    expect(style.width).toEqual(200);
    expect(elementProps.shadowColor).toMatch('black');
    expect(elementProps.shadowOffset).toEqual({width: 4, height: 8});
    expect(elementProps.shadowOpacity).toEqual(0.5);
    expect(elementProps.shadowRadius).toEqual(8);
    expect(elementProps.elevation).toEqual(8);
    expect(toJSON()).toMatchSnapshot();
  });

  test('color', () => {
    const props = {
      testID: 'my-view',
      width: 200,
      height: 100,
      bg: '#FECF28',
      borderWidth: 1,
      borderColor: 'black',
      borderStyle: 'solid',
    };
    const {queryByTestId, toJSON} = render(<View {...props}> </View>);
    const style = queryByTestId('my-view').props.style[0];
    expect(style.backgroundColor).toMatch('#FECF28');
    expect(style.borderColor).toMatch('black');
    expect(style.borderStyle).toMatch('solid');
    expect(style.borderWidth).toEqual(1);
    expect(style.height).toEqual(100);
    expect(style.width).toEqual(200);
    expect(toJSON()).toMatchSnapshot();
  });
});

// TODO: add more case
// TODO: add negative case
