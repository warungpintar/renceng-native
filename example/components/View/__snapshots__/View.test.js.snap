// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View basic 1`] = `
<View
  bg="white"
  borderColor="black"
  borderStyle="solid"
  borderWidth={1}
  height={100}
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderColor": "black",
        "borderStyle": "solid",
        "borderWidth": 1,
        "height": 100,
        "width": 200,
      },
      undefined,
    ]
  }
  testID="my-view"
  width={200}
>
   
</View>
`;

exports[`View color 1`] = `
<View
  bg="#FECF28"
  borderColor="black"
  borderStyle="solid"
  borderWidth={1}
  height={100}
  style={
    Array [
      Object {
        "backgroundColor": "#FECF28",
        "borderColor": "black",
        "borderStyle": "solid",
        "borderWidth": 1,
        "height": 100,
        "width": 200,
      },
      undefined,
    ]
  }
  testID="my-view"
  width={200}
>
   
</View>
`;

exports[`View onPress 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="my-view"
>
   
</View>
`;

exports[`View rounded 1`] = `
<View
  bg="white"
  borderColor="black"
  borderRadius={8}
  borderStyle="solid"
  borderWidth={1}
  height={100}
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderColor": "black",
        "borderRadius": 8,
        "borderStyle": "solid",
        "borderWidth": 1,
        "height": 100,
        "width": 200,
      },
      undefined,
    ]
  }
  testID="my-view"
  width={200}
>
   
</View>
`;

exports[`View shadow 1`] = `
<View
  bg="white"
  elevation={8}
  height={100}
  shadowColor="black"
  shadowOffset={
    Object {
      "height": 8,
      "width": 4,
    }
  }
  shadowOpacity={0.5}
  shadowRadius={8}
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "height": 100,
        "width": 200,
      },
      undefined,
    ]
  }
  testID="my-view"
  width={200}
>
   
</View>
`;
